library/src/interactor_impl.cxx

```diff
@@ -167,9 +167,19 @@ class interactor_impl::internals
     bool checkColoring = false;
     bool render = false;
 
-    // Available keycodes: W
+    // Available keycodes: None
     switch (keyCode)
     {
+      case 'W':
+        int animationIndex;
+        self->AnimationManager->CycleAnimation();
+        animationIndex = self->AnimationManager->GetAnimationIndex();
+        self->Options.set("scene.animation.index", animationIndex);
+        self->AnimationManager->SelectAnimationIndex(animationIndex);
+        self->Options.set("ui.animationname-info", self->AnimationManager->GetAnimationName());
+        self->AnimationManager->LoadAtTime(0);
+        render = true;
```

Comment:
imo this should be as simple as:

```
       self->AnimationManager->CycleAnimation();
       animationIndex = self->AnimationManager->GetAnimationIndex();
       self->Options.set("scene.animation.index", animationIndex);
       render = true;
```

No need to call SelectAnimationIndex because animationManager should already be aware
No need to call LoadAtTime(0), because the animationManager should take care of it (and even keep the animation at the right time step if it makes sense)

Reply:
Thanks! I am learning what belongs in the frontend vs. backend. Slowly, as I am working more on this project. How about the display of the current animation name? I wasn't quite sure where to place that.

Reply:
> How about the display of the current animation name? 

It should be displayed on the CheatSheet instead of the filename actor.
You can look at how the scalars are handled in the cheatsheet:

https://github.com/f3d-app/f3d/blob/4bd4e38592989875186f5fa0460a214887201b2e/library/VTKExtensions/Rendering/vtkF3DRendererWithColoring.cxx#L923

Press H to show it.

Reply:
Okay, sounds good. Do I need to create `vtkF3DRendererWithAnimation.cxx`? I'm unsure which file I should place my display of animation name within.`vtkF3DRenderer.cxx` does not have access to importer while `vtkF3DRendererWithColoring.cxx` only displays for models with coloring. I'm also unsure how or if I can use animationManager within `vtkF3DRenderer.cxx`. 

Help appreciated!

Reply:
@mwestphal Hey, I think you missed this message yesterday when you were sharing feedback. I'm struggling to figure out how to display my animation name within the cheatsheet menu. Help appreciated!

Reply:
You simply need to add a line about the animation name in vtkF3DRendererWithColoring::FillCheatSheetHotkeys

Reply:
If I write the code in there, it will only display for models with coloring. For instance, it won't display for models like this one https://sketchfab.com/3d-models/caretaker-animated-e05da99d58ed4013ab1424d6895d56e8

Reply:
`vtkF3DRenderer::FillCheatSheetHotkeys` then :)

Reply:
Conversation moved to F3D Discord. Used this patch https://github.com/f3d-app/f3d/commit/de2302875e5905775ffbfb66b1be71e8bef0db70